Tutorial: Replace Hotjar, use session recordings for free (#2995)
PostHog/posthog.com: contents/tutorials/free-hotjar-alternative.md

[Hotjar is also costly on its own](https://www.hotjar.com/pricing/), often costing hundreds of pounds per month or limiting free users to only 1,050 monthly session recordings. 

In this tutorial we’ll explore how PostHog can be used as an alternative to Hotjar, offering greater functionality so that teams don’t need to use multiple tools at once.

## Step 1: Get started with PostHog

![free hotjar alternative - start](../images/tutorials/hotjar-alternative/get-started-with-posthog.png)

If you haven’t already started using PostHog, that’s the obvious first step. You can choose between self-hosted or cloud-hosted options depending on your preferences and privacy requirements. Broadly speaking self-hosted deployments require more technical knowledge to get started, but offer greater control over your data — but [check the Pricing page for more information](/pricing).

Whichever option and pricing tier you choose, PostHog offers a generous free allowance of events:

> - Users on [PostHog Cloud](/pricing) receive 10,000 free events each month

Reviewer: ```suggestion
- Users on [PostHog Cloud](/pricing) receive 1 million free events each month
```

> - Users on [PostHog Scale](/pricing) receive 1 million free events each month

Reviewer: ```suggestion
- Users on [PostHog Scale](/pricing) also receive 1 million free events each month
```

- Users on [PostHog Open Source](/pricing) have no limit on the number of tracked events

It’s worth noting that some tiers do not have access to premium features, such as [Correlation Analysis](/docs/user-guides/correlation) or [A/B Experimentation](/docs/user-guides/experimentation). However, _all_ tiers have access to core features such as [Session Recording](/docs/user-guides/recordings), [Funnel](/docs/user-guides/funnels) & [Trend Analysis](/docs/user-guides/trends) and [Feature Flags](/docs/user-guides/feature-flags). 

> Session Recordings do not currently count towards your free event limit and there is no limit on the number of recordings you can capture on any pricing tier.

## Step 2: Enable event capture

![free hotjar alternative - disable auto](../images/tutorials/hotjar-alternative/disable-autocapture.png)

We strongly recommend enabling PostHog's autocapture feature to make sure you get the most out of Session Recording and that common events, such as pageviews, are automatically ingested. You can then analyse these events and filter Session Recordings for them, combining the capabilities of Hotjar with a product analytics platform. 

> To start capturing events you must add the snippet found in your Project Settings page above the `</head>` tag in your website HTML. This snippet also enables you to capture custom events which may be unique to your product.

Author: ```suggestion
To start capturing events you must add the snippet found in your Project Settings page above the `</head>` tag in your website HTML. This snippet also enables you to capture custom events, which you can define in the Events & Actions page. You can then filter any captured session based on these [custom events](/docs/user-guides/events), or even [use Feature Flags to _only_ capture these sessions](/docs/user-guides/recordings#further-controls). 
```

If you haven’t already started using PostHog, that’s the obvious first step. You can choose between self-hosted or cloud-hosted options depending on your preferences and privacy requirements. Broadly speaking self-hosted deployments require more technical knowledge to get started, but offer greater control over your data — but [check the Pricing page for more information](/pricing).

Whichever option and pricing tier you choose, PostHog offers a generous free allowance of events:

- Users on [PostHog Cloud](/pricing) receive 10,000 free events each month
- Users on [PostHog Scale](/pricing) receive 1 million free events each month
- Users on [PostHog Open Source](/pricing) have no limit on the number of tracked events

It’s worth noting that some tiers do not have access to premium features, such as [Correlation Analysis](/docs/user-guides/correlation) or [A/B Experimentation](/docs/user-guides/experimentation). However, _all_ tiers have access to core features such as [Session Recording](/docs/user-guides/recordings), [Funnel](/docs/user-guides/funnels) & [Trend Analysis](/docs/user-guides/trends) and [Feature Flags](/docs/user-guides/feature-flags). 

> > Session Recordings do not currently count towards your free event limit and there is no limit on the number of recordings you can capture on any pricing tier.

Reviewer 2: My concern with stating this - how long will we keep it free for? Session recordings is pretty expensive for us to operate and if we get a few people abusing this we'll have to put in some kind of limit - fairly quickly.

Also there's a retention limit of 29days I believe (which might be worth calling out somewhere) @rcmarron

Author: I get this, but I'm unaware of any definite plans to begin charging for it or limiting it at the moment - and we've recognised this loophole at least since early November, when it was first discussed. 

At the moment it does exist without limits or charges, so I personally feel we should either be closing that loophole or taking advantage of it. If we're not planning to close it...

I'll add a line about retention policy in there.

Author: Sod it, pushed a quick update to reflect this.

> ## Step 2: Disable autocapture (optional)

Reviewer 2: Given we say this above:

> limitations of Hotjar is that it offers only a handful of such tools and doesn’t provide the deeper product analytics tools

I feel its counter productive to encourage people to turn off any event tracking, since we believe the value comes from having both session recording and product analytics.

I appreciate it makes the article less powerful, I would advocate for saying "get started with session recordings for free...and you don't even need to ingest any events...", but not actively encourage people to turn off what is potentially valuable data.

Also the session recordings experience is no-where near as good without events, because you can't navigate through the session recordings or find interesting ones easily - so we might find that people ultimately have a poor experience, without any events.

Autocaptured events count towards your total number of events, so if you wish to stay within your free allowance then it may be best to [disable autocapture](https://posthog.com/docs/integrate/client/js#config). 

Users who opt out of capturing _any_ events will not be able to record sessions as, while session recordings do not count towards your allowance of free events, each recording does generate at least one event. 

> Disabling autocapture will mean you can only track custom events which you have defined within PostHog, and will greatly limit your ability to analyze product usage.

## Step 3: Enable Session Recording & Heatmaps

![free hotjar alternative - watch replay](../images/tutorials/hotjar-alternative/enable-recording.png)

Session Recording is not enabled by default, so must be turned on to start capturing sessions.. Currently, recordings can only be used with [PostHog’s JavaScript library](/docs/integrate/client/js) and must be enabled within Project Settings. 

> Once Session Recordings are enabled, you must add the sites you want to record sessions from to your list of permitted domains and URLs. This list is also found in your Project Settings.

Author: ```suggestion
Once Session Recordings are enabled, you must add the sites you want to record sessions from to your list of permitted domains and URLs. This list is also found in your Project Settings. Session Recordings are currently retained for 29 days, before deletion. 
```

@rcmarron
